### PR TITLE
Create new stream to miner when it's needed

### PR DESCRIPTION
**Pull request by ilyk**
while we're not using newly created stream to the miner not in the `GetPaych` nor `AllocateLane` calls, we shouldn't create it here. Let's create it when it will be useful already.

that should help to prevent stream reset error.

### CODE REVIEW

File: retrieval/client.go [hunk 1]
```diff
@@ -98,12 +98,6 @@ type clientStream struct {
 // > DealProposal(...)
 // < ...
 func (c *Client) RetrieveUnixfs(ctx context.Context, root cid.Cid, size uint64, total types.BigInt, miner peer.ID, client, minerAddr address.Address, out io.Writer) error {
-	s, err := c.h.NewStream(ctx, miner, ProtocolID)
-	if err != nil {
-		return xerrors.Errorf("failed to open stream to miner for retrieval query: %w", err)
-	}
-	defer s.Close()
-
```
Comment (magik6k) [inline retrieval/client.go]: We should make sure that we can actually talk to the peer - simplest way would be to use libp2p ping. For one example, see
https://github.com/ipfs/go-ipfs/blob/master/core/commands/ping.go#L106

Comment (ilyk) [reply retrieval/client.go]: Please check this out

**Comment by magik6k** [comment]
Tests should start to pass after https://github.com/filecoin-project/lotus/pull/980 is merged

**Comment by ipervushyn** [comment]
> Tests should start to pass after #980 is merged

how to re-run test?

**Comment by ilyk** [comment]
> Tests should start to pass after #980 is merged

how to re-run a test?

> Tests should start to pass after #980 is merged

how to re-run a test

**Comment by whyrusleeping** [comment]
It looks like theres not actually any changes here anymore. Did a different PR get merged? Should we close this?

**Comment by magik6k** [comment]
This is either merged already somehow, or github is doing something funny. Feel free to open a new PR if something from here could still be used

**Pull request closed without merging**